### DISCUSSION TITLE
Endless request

### DISCUSSION THREAD

**Discussion by butulkus** [aptos-labs/aptos-developer-discussions#266] [Questions]
### Describe your question in detail.

Can't get a result on line `const tokens = await aptos.getChainId();` - endless loading.
Code: 
```
import { Aptos } from "@aptos-labs/ts-sdk";

async function aptos() {
    const aptos = new Aptos();
    const tokens = await aptos.getChainId();
    console.log(tokens);
}

aptos().then(_ => console.log('finished'))
```

ts debugging:
<img width="1063" alt="image" src="https://github.com/aptos-labs/aptos-developer-discussions/assets/96844628/a703052e-4589-4888-9a08-03cfae479861">

react debugging:
<img width="630" alt="image" src="https://github.com/aptos-labs/aptos-developer-discussions/assets/96844628/1b257c45-30e7-4ba6-b101-93286a897950">

differect result with same function call `await aptos.getChainId();`

### What have you tried or looked at?  Or how can we reproduce the error?

It's fun, but using react-ts requests to library work fine(used `npm create vite@latest`)

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

### Describe your environment or tooling in detail

macos
tsconfig.json:
```
{
  "compilerOptions": {
    "resolveJsonModule": true,
    "sourceMap": true,
    "target": "es2016",
    "module": "commonjs",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true
  }
}
```

package.json:
```
{
  "dependencies": {
    "@aptos-labs/ts-sdk": "^1.17.0",
    "@types/node": "^20.12.12",
    "ethers": "^6.12.1",
    "googleapis": "^137.1.0",
    "ts-node": "^10.9.2",
    "typescript": "^5.4.5"
  }
}

```

**Comment by BriungRi** [comment]
Is this happening consistently? There is a chance that this function was being run against devnet while devnet was being redeployed (it gets redeployed on roughly a weekly basis)

**Reply by butulkus** [reply]
Consistently. I've been trying to launch it since last week. It's funny but inside React porject it works well, i think problem with project set up, so i've added tsconfig + package.json to discussion.

**Reply by gregnazario** [reply]
Underneath the TypeScript call, it's just a call to this REST API https://fullnode.devnet.aptoslabs.com/v1 and then pulls the ChainId from there. I agree I think the only thing that could happen is either, it was being deployed at the time you requested it, or some network issue on your end.

```
{
"chain_id": 139,
"epoch": "49",
"ledger_version": "51020612",
"oldest_ledger_version": "0",
"ledger_timestamp": "1718647967215692",
"node_role": "full_node",
"oldest_block_height": "0",
"block_height": "8878533",
"git_hash": "d76b5bb423b78b2b9affc72d3853f0d973d3f11f"
}
```

**Reply by butulkus** [reply]
Is there a chance you check it by creating an typescript application with aptos sdk?
